go/inst/instance_binlog_dao.go

```diff
@@ -619,68 +619,158 @@ func getNextBinlogEventsChunk(instance *Instance, startingCoordinates BinlogCoor
 	return events, err
 }
 
-// GetNextBinlogCoordinatesToMatch is given a twin-coordinates couple for a would-be replica (instanceKey) and another
-// instance (otherKey).
+// rpad formats the binlog coordinates to a given size. If the size
+// increases this value is modified so it can be reused later. This
+// is to ensure consistent formatting in debug output.
+func rpad(coordinates BinlogCoordinates, length *int) string {
+	s := fmt.Sprintf("%+v", coordinates)
+	if len(s) > *length {
+		*length = len(s)
+	}
+
+	if len(s) >= *length {
+		return s
+	}
+	return fmt.Sprintf("%s%s", s, strings.Repeat(" ", *length-len(s)))
+}
+
+// used by GetNextBinlogCoordinatesToMatch to format debug information appropriately
+// format the event information in debug output
+func formatEventCleanly(event BinlogEvent, length *int) string {
+	return fmt.Sprintf("%+v %+v; %+v", rpad(event.Coordinates, length), event.EventType, strings.Split(strings.TrimSpace(event.Info), "\n")[0])
+}
+
+// Only do special filtering if instance is MySQL-5.7 and other
+// is MySQL-5.6 and in pseudo-gtid mode.
+func special56To57filterProcessing(instance *Instance, other *Instance) (bool, bool, error) {
+	// be paranoid
+	if instance == nil || other == nil {
+		return false, false, fmt.Errorf("special56To57filterProcessing: instance or other is nil. Should not happen")
+	}
+
+	filterInstance := instance.FlavorNameAndMajorVersion() == "MySQL-5.7" && // 5.7 replica
+		other.FlavorNameAndMajorVersion() == "MySQL-5.6" // replicating under 5.6 master
+
+	// The logic for other is a bit weird and may require us
+	// to check the instance's master.  To avoid this do some
+	// preliminary checks first to avoid the "master" access
+	// unless absolutely needed.
+
+	if instance.LogBinEnabled || // instance writes binlogs (not relay logs)
+		instance.FlavorNameAndMajorVersion() != "MySQL-5.7" || // instance NOT 5.7 replica
+		other.FlavorNameAndMajorVersion() != "MySQL-5.7" { // new master is NOT  5.7
```

Comment:
I'm curious, why `MySQL-5.6` and `MySQL-5.7` as opposed to `5.6` and `5.7`? Does the problem not happen with Percona `5.6` & `5.7`?